src/test/registration.test.tsx

```diff
@@ -0,0 +1,40 @@
+// registration.test.tsx
+import { fireEvent, render, waitFor } from '@testing-library/react';
+import { describe, expect, it } from 'vitest';
+import { Registration } from '../pages/registration/registration';
+
```

Comment:
```suggestion
import { MemoryRouter } from 'react-router-dom';
import React from 'react';
import type { RenderResult } from '@testing-library/react';

const renderInsideRouter = (testedElement: React.ReactElement): RenderResult => {
  return render(<MemoryRouter>{testedElement}</MemoryRouter>);
};
```
These imports are needed to create the router wrapper.
And here is the wrapper creation function itself.

Reply:
Definitely this option is better and the tests work